### ISSUE TITLE
DataTable: filters are not rerendered when their state is changed

### ISSUE THREAD

**Issue by VsevolodGolovanov** [primefaces/primereact#1352]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

https://codesandbox.io/s/primereact-test-0c3wq

**Current behavior**
When changing the controlled state of DataTable filters, the changes affect displayed data in the table, but the filter fields themselves are not updated.

**Expected behavior**
The filter fields are updated to show their current state.

**Minimal reproduction of the problem with instructions**
Open the linked codesandbox.
Type something in both the column filter and the global filter.
Press the reset button.
Notice how data in the table is unfiltered and visible again, but filter fields still show the values that you typed in.

![image](https://user-images.githubusercontent.com/8070317/80581948-99a71900-8a16-11ea-83e4-e70bedc78c83.png)

**Please tell us about your environment:**
Win10, WebStorm.

* **React version:**
16.13.1, 16.8.0.

* **PrimeReact version:**
4.1.2, 4.2.0.

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Firefox 75.
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by mertsincan** [comment]
To reset globalFilter input, please use;
```
...
    const header = (
        <>
        <InputText
            type="search"
            value={this.state.globalFilter}
            onChange={e => this.setState({ globalFilter: e.currentTarget.value })}
            placeholder="Global filter"
            size={30}
        />
        <Button
            label="Reset all filters"
            onClick={() => this.setState({ filters: {}, globalFilter: "" })}
        />
        </>
    );
...
```

**Comment by VsevolodGolovanov** [comment]
Ah, I forgot to bind InputText's value. My mistake.

**Comment by andhure** [comment]
Is there a way we can reset filter text box of individual columns using a reset function? I understand that it is possible for global filters.

**Comment by andhure** [comment]
Got it. It is fixed in 4.2.1. Thank you
